config/config.go

```diff
@@ -0,0 +1,655 @@
+	GrandpaAuthority bool               `mapstructure:"grandpa-authority"`
+	WasmInterpreter  string             `mapstructure:"wasm-interpreter,omitempty"`
+	GrandpaInterval  time.Duration      `mapstructure:"grandpa-interval,omitempty"`
+}
+
+// StateConfig contains the configuration for the state.
+type StateConfig struct {
+	Rewind uint `mapstructure:"rewind,omitempty"`
+}
+
+// RPCConfig is to marshal/unmarshal toml RPC config vars
+type RPCConfig struct {
+	RPCExternal       bool     `mapstructure:"rpc-external,omitempty"`
+	UnsafeRPC         bool     `mapstructure:"unsafe-rpc,omitempty"`
+	UnsafeRPCExternal bool     `mapstructure:"unsafe-rpc-external,omitempty"`
+	Port              uint32   `mapstructure:"port,omitempty"`
+	Host              string   `mapstructure:"host,omitempty"`
+	Modules           []string `mapstructure:"modules,omitempty"`
+	WSPort            uint32   `mapstructure:"ws-port,omitempty"`
+	WSExternal        bool     `mapstructure:"ws-external,omitempty"`
+	UnsafeWSExternal  bool     `mapstructure:"unsafe-ws-external,omitempty"`
+}
+
+// PprofConfig contains the configuration for Pprof.
+type PprofConfig struct {
+	Enabled          bool   `mapstructure:"enabled,omitempty"`
+	ListeningAddress string `mapstructure:"listening-address,omitempty"`
+	BlockProfileRate int    `mapstructure:"block-profile-rate,omitempty"`
+	MutexProfileRate int    `mapstructure:"mutex-profile-rate,omitempty"`
+}
+
+// ValidateBasic does the basic validation on BaseConfig
+func (b *BaseConfig) ValidateBasic() error {
+	if b.Name == "" {
+		return fmt.Errorf("name cannot be empty")
+	}
+	if b.ID == "" {
+		return fmt.Errorf("id cannot be empty")
+	}
+	if b.BasePath == "" {
+		return fmt.Errorf("base-path directory cannot be empty")
+	}
+	if b.ChainSpec == "" {
+		return fmt.Errorf("chain-spec cannot be empty")
+	}
+	if b.PrometheusPort == "" {
+		return fmt.Errorf("prometheus port cannot be empty")
+	}
+	if uint32Max < b.RetainBlocks {
+		return fmt.Errorf(
+			"retain-blocks value overflows uint32 boundaries, must be less than or equal to: %d",
+			uint32Max,
+		)
+	}
+
+	return nil
+}
+
+// ValidateBasic does the basic validation on LogConfig
+func (l *LogConfig) ValidateBasic() error {
+	return nil
+}
+
+// ValidateBasic does the basic validation on AccountConfig
+func (a *AccountConfig) ValidateBasic() error {
+	if a.Key == "" {
+		return fmt.Errorf("key cannot be empty")
+	}
+	//if a.Unlock == "" {
+	//	return fmt.Errorf("unlock cannot be empty")
+	//}
+
+	return nil
+}
+
+// ValidateBasic does the basic validation on NetworkConfig
+func (n *NetworkConfig) ValidateBasic() error {
+	if n.Port == 0 {
+		return fmt.Errorf("port cannot be empty")
+	}
+	//if n.ProtocolID == "" {
+	//	return fmt.Errorf("protocol cannot be empty")
+	//}
+	//if n.MinPeers == 0 {
+	//	return fmt.Errorf("minimum-peers cannot be empty")
+	//}
+	//if n.MaxPeers == 0 {
+	//	return fmt.Errorf("maximum-peers cannot be empty")
+	//}
```

Comment:
why are these commented out?

Reply:
I need help with the default config values before enabling the validation